### ISSUE TITLE
Inconsistent Generated Code

### ISSUE THREAD

**Issue by geisterfurz007** [widgetbook/widgetbook#935]
**Description**

We are using a CI job to ensure that all generated files that were commited to VCS are up-to-date by running build_runner and diffing the state. This allows us to run a much faster CI because job don't have to wait 30 seconds for build_runner to finish while still having the safety of knowing the generated code is up to date.

Twice now, this strategy has failed because the CI ended up generating different code than the developers on their machine, despite all versions being identical. Worse even, when manually doing the changes shown in the diff and pushing that, a diff is shown that either requests what was previously there or something entirely different.

This causes our CI to be unstable and flaky and us as developers losing the safety of knowing the generated code is correct.

**Steps To Reproduce**

I am unfortunately not sure how to reproduce this properly which I understand makes things difficult. I would love to have a look around the generator code however to see if I can find a place to sneak in a tactical `sort` somewhere to remedy this issue.

**Expected Behavior**
Every run of widgetbook on the same codebase results in the same widgetbook.directories.g.dart (or whatever is configured).

**Screenshots**

<details>
<summary>In the first failed pipeline, the CI reports a generated diff that moves the `bottom_navigation_bar` from above the `antrag_card` to above the `heading`:</summary>

![grafik](https://github.com/widgetbook/widgetbook/assets/26303198/eea6dcca-c08f-4d8f-97a5-a35366f146ae)

</details>

<details>
<summary>So this commit was pushed:</summary>

![grafik](https://github.com/widgetbook/widgetbook/assets/26303198/1454b97c-54ac-4a1e-b41b-6e57d3147f21)
</details>

<details>
<summary>And this was the resulting CI log requesting the `bottom_navigation_bar` be moved back to where it was before:</summary>

![grafik](https://github.com/widgetbook/widgetbook/assets/26303198/88e24347-9eee-4128-8d6e-161dc4cf9f4e)
</details>

**Additional Context**
The problem only showed up after a new import was introduced it appeared like but that might have been unrelated.

I would love to give this a shot btw!

**Comment by geisterfurz007** [comment]
Oh no... Oh no, oh noes, this one might turn out embarassing for me actually... We had indeed updated to widgetbook@3.2.0 when this problem first came up and then the problem went away. I just realized that we did *not* update widgetbook_generator to 3.2.0. 

With the update of the generator, this now looks very alphabetically sorted indeed and the CI seems to agree as well.

I apologize for the ruckus caused! I hope I can find another issue to work on to make up for it :)
